pbj-core/pbj-runtime/src/main/java/com/hedera/pbj/runtime/io/stream/ReadableStreamingData.java

```diff
@@ -27,14 +31,36 @@ public class ReadableStreamingData implements ReadableSequentialData, AutoClosea
     private boolean eof = false;
 
     /**
-     * Creates a {@code FilterInputStream} that implements {@code DataInput} API.
+     * Creates a new streaming data object on top of a given input stream.
      *
      * @param in the underlying input stream, can not be null
      */
     public ReadableStreamingData(@NonNull final InputStream in) {
         this.in = requireNonNull(in);
     }
 
+    /**
+     * Opens a new input stream to read a given file and creates a new streaming data object on top
+     * of this stream.
+     *
+     * @param file the file, can not be null
+     * @throws IOException if an I/O error occurs
+     */
+    public ReadableStreamingData(@NonNull final Path file) throws IOException {
+        this.in = Files.newInputStream(file, StandardOpenOption.READ);
+        this.limit = Files.size(file);
```

Comment:
Would it be worth checking here if `file` is a "regular" file?
We should never be trying to read from any other type, but given `Files.size` is undefined for any non-regular file (e.g. pipes, sockets, devices, etc...), it might be worth checking.

Reply:
OK, I added a check that the provided file is a regular file (sym links followed), and the file is readable.